ledger/participant-integration-api/src/main/scala/platform/apiserver/update/UpdateMapper.scala

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.update
+
+import com.daml.ledger.api.domain.{ParticipantParty, User}
+import com.daml.ledger.participant.state.index.v2._
+import com.daml.lf.data.Ref
+import com.google.protobuf.field_mask.FieldMask
+import com.daml.platform.apiserver.update.UpdatePathsTrie.MatchResult
+
+object UpdateMapper {
+
+  private val userAnnotationsPath = List(
+    FieldNames.UpdateUserRequest.user,
+    FieldNames.User.metadata,
+    FieldNames.Metadata.annotations,
+  )
+  private val userPrimaryPartyPath =
+    List(FieldNames.UpdateUserRequest.user, FieldNames.User.primaryParty)
+  private val userIsDeactivatePath =
+    List(FieldNames.UpdateUserRequest.user, FieldNames.User.isDeactivated)
+
+  private val fullUpdateUserTree = UpdatePathsTrie
+    .fromPaths(
+      Seq(
+        UpdatePath(userAnnotationsPath, modifier = UpdatePathModifier.NoModifier),
+        UpdatePath(userPrimaryPartyPath, modifier = UpdatePathModifier.NoModifier),
+        UpdatePath(userIsDeactivatePath, modifier = UpdatePathModifier.NoModifier),
+      )
+    )
+    .getOrElse(sys.error("Failed to create full update user tree. This should never happen"))
+
+  private def noUpdate[A]: Result[Option[A]] = Right(None)
+
+  /** Validates its input and produces an update object.
+    * NOTE: The return update object might represent an empty (no-op) update.
+    *
+    * @param user represents the new values for the update
+    * @param updateMask indicates which fields should get updated
+    */
+  def toUserUpdate(user: User, updateMask: FieldMask): Result[UserUpdate] = {
```

Comment:
Where is this method going to be used ultimately? Currently it is only used in the unit test.

Reply:
It will bridge the data representation from the request (user + updateMask) and convert it to 'userUpdate' consumed by the store layer.
It also validates the updateMask

Reply:
It'll be used in the ApiService layer